### PR TITLE
Core list_tables fetches also the database info

### PR DESCRIPTION
**Pull request by PopDaph**
Fetch the database data along with fetching its table info. 
Required for https://github.com/dust-tt/dust/pull/2931

### CODE REVIEW

File: core/src/stores/postgres.rs [hunk 1]
```diff
@@ -2345,7 +2345,7 @@ impl Store for PostgresStore {
         data_source_id: &str,
         database_id: &str,
         limit_offset: Option<(usize, usize)>,
-    ) -> Result<(Vec<DatabaseTable>, usize)> {
+    ) -> Result<(Option<Database>, Vec<DatabaseTable>, usize)> {
```
Comment (fontanierh) [inline core/src/stores/postgres.rs:2348]: Why is database option here ?

**Review by fontanierh — CHANGES_REQUESTED** [review]
I think it shouldn't be an option

File: core/src/stores/postgres.rs (hunk 1 above)
Comment (PopDaph) [reply core/src/stores/postgres.rs]: Indeed it shouldn't, I fixed it!

**Review by fontanierh — APPROVED** [review]

File: core/src/stores/postgres.rs [hunk 2]
```diff
@@ -2367,17 +2367,35 @@ impl Store for PostgresStore {
             _ => unreachable!(),
         };
 
-        // get the database row id
+        // get the database row
         let r = c
             .query(
-                "SELECT id FROM databases WHERE data_source = $1 AND database_id = $2 LIMIT 1",
+                "SELECT id, databases.created, databases.database_id, databases.name \
```
Comment (spolu) [inline core/src/stores/postgres.rs:2373]: Nit: We don’t need the “databases.” prefix?

**Review by spolu — APPROVED** [review]

**Pull request merged**